### PR TITLE
Fix proxyconfig annotation lead to kube-inject panic

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
MeshConfig is nil if injector exists. Set it to default proxyconfig to avoid of nil pointer panic.

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
can you elaborate more when this can happen

File: pkg/config/mesh/mesh.go
Comment (howardjohn) [inline pkg/config/mesh/mesh.go:144]: how does clone return `nil`? is `meshConfig` already nil? If so, seems like the check should be before clone?

Also it seems odd that we will make a empty mesh config. in 99% of places we use DefaultMeshConfig.

This seems like we should be changing the place that _calls_ ApplyProxyConfig, instead of ApplyProxyConfig itself?

**Comment by hanxiaop** [comment]
> can you elaborate more when this can happen

@hzxuzhonghu @howardjohn When using `kube-inject` cmd without providing the injector file, it will use the webhook injector and return nil meshconfig. In this case, when a pod has the annotation `proxy.istio.io/config`, the `ApplyProxyConfig` func will panic with the nil meshconfig. I think John is right, and I just changed the code place instead of revising the `ApplyProxyConfig` itself.

**Review by howardjohn — APPROVED** [review]
should we add a test?

**Review by howardjohn — APPROVED** [review]

**Comment by intel-dlanders** [comment]
What's the process here? This bug breaks your blog post on [proxyless grpc](https://istio.io/latest/blog/2021/proxyless-grpc/#deploy-the-application). It looks like we're waiting for one reviewer approval. Can we get it?! Please.

**Review by irisdingbj — APPROVED** [review]
Thanks for the fix!

**Comment by GregHanson** [comment]
@hanxiaop does this need to be cherrypicked to 1.14 or 1.15 release branches?

**Pull request merged**
